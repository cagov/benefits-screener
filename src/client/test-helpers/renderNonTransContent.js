import React from "react";
import { shallow } from "enzyme";
import { I18nextProvider, Trans } from "react-i18next";
import i18n from "./i18nForTests";
import shallowUntilTarget from "./shallowUntilTarget";

function renderNonTransContent(Component, props = {}) {
  const providersWrapper = shallow(
    <I18nextProvider i18n={i18n}>
      <Component {...props} />
    </I18nextProvider>
  );

  const wrapper = shallowUntilTarget(providersWrapper, Component.name);
  // Incorrectly using renderNonTransContent to render a component that
  // contains a Trans component results in a silent failure (the rendered
  // snapshot will contain the Trans component itself instead of the rendered
  // contents). This test makes that silent failure more obvious.
  expect(wrapper.containsMatchingElement(<Trans />)).toBe(false);

  return wrapper;
}

module.exports = renderNonTransContent;
