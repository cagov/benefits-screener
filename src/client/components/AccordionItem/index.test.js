import renderNonTransContent from "../../test-helpers/renderNonTransContent";
import Component from "./index";

describe("<AccordionItem />", () => {
  it("renders the component", async () => {
    const wrapper = renderNonTransContent(Component, "AccordionItem");

    expect(wrapper).toMatchSnapshot();
  });
});
