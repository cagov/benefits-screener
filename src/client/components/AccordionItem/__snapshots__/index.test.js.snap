// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionItem /> renders the component 1`] = `
<Fragment>
  <Alert
    className="d-flex toggleAccordion"
    closeLabel="Close alert"
    onClick={[Function]}
    show={true}
    transition={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "appear": false,
          "in": false,
          "mountOnEnter": false,
          "timeout": 300,
          "unmountOnExit": false,
        },
        "displayName": "Fade",
        "render": [Function],
      }
    }
    variant="secondary"
  >
    <button
      aria-label="Show details for this week"
    >
      <div
        className="flex-fill"
      >
        <span
          className="toggleCharacter"
        >
          â€“
        </span>
        <div>
          header react element
        </div>
      </div>
    </button>
  </Alert>
  <div
    className="detail"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div>
      detailed body react element
    </div>
  </div>
</Fragment>
`;
